packages/typespec-python/test/azure/mock_api_tests/test_payload_multipart_file_array.py

```diff
@@ -0,0 +1,104 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import Dict, Any
+from pathlib import Path
+import pytest
+from payload.multipart import MultiPartClient, models
+from payload.multipart._model_base import Model
+
+JPG = Path(__file__).parent.parent.parent / "generic_mock_api_tests/data/image.jpg"
+PNG = Path(__file__).parent.parent.parent / "generic_mock_api_tests/data/image.png"
+
+
+@pytest.fixture
+def client():
+    with MultiPartClient(endpoint="http://localhost:3000") as client:
+        yield client
+
+
+@pytest.mark.parametrize(
+    "op_name,model_class,data,file",
+    [
+        (
+            "binary_array_parts",
+            models.BinaryArrayPartsRequest,
+            {"id": "123"},
+            {"pictures": [PNG, PNG]},
+        ),
+        (
+            "complex",
+            models.ComplexPartsRequest,
+            {"id": "123", "previousAddresses": [models.Address(city="Y"), models.Address(city="Z")], "address": models.Address(city="X")},
+            {"pictures": [PNG, PNG], "profileImage": JPG},
+        )
+    ],
+)
+def test_multi_part(client: MultiPartClient, op_name, model_class, data, file):
+    file_bytes = lambda p: open(str(p), "rb").read()
+    op = getattr(client.form_data, op_name)
+    # test bytes
+    body = {k: ([file_bytes(vi) for vi in v] if isinstance(v, list) else file_bytes(v)) for k, v in file.items()}
+    body.update(data)
+    op(body)
+    op(model_class(body))
+
+    # test io
+    file_io = lambda p: open(str(p), "rb")
+    body = {k: ([file_io(vi) for vi in v] if isinstance(v, list) else file_io(v)) for k, v in file.items()}
+    body.update(data)
+    op(body)
+
+    body = {k: ([file_io(vi) for vi in v] if isinstance(v, list) else file_io(v)) for k, v in file.items()}
+    body.update(data)
+    with pytest.raises(TypeError):
+        # caused by deepcopy when DPG model init
+        op(model_class(body))
```

Comment:
DPG model will raise error when deepcody of init if property value is io. @iscai-msft